File: android/app/src/main/kotlin/com/octomed/octo360/event/EventHandler.kt
package com.octomed.octo360.event;

import android.os.Handler
import android.os.Looper
import io.flutter.plugin.common.EventChannel
import java.util.*

class EventHandler : EventChannel.StreamHandler {
    private var eventSink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
    }

//    fun send(task: BioTask) {
//        Handler(Looper.getMainLooper()).post {
//            eventSink?.success(mapOf(
//                "event" to task.method,
//                "body" to task.data
//            ))
//        }
//    }

    fun send(event: String, body: Map<String, Any>?) {
        Handler(Looper.getMainLooper()).post {
            eventSink?.success(mapOf("event" to event, "body" to body))
        }
    }

    fun send(event: String, body: ArrayList<HashMap<String, String>>) {
        Handler(Looper.getMainLooper()).post {
            eventSink?.success(mapOf("event" to event, "body" to body))
        }
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }
}

File: ios/Runner/event/EventHandler.swift
//
//  EventHandler.swift
//  Runner
//
//  Created by Manh Tran on 25/09/2023.
//

import Foundation

class EventHandler: NSObject, FlutterStreamHandler {
    private var eventSink: FlutterEventSink?
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        self.eventSink = nil
        return nil
    }
    
    
    func send(event: String, body: [String: Any]?) {
        eventSink?(["event": event, "body": body])
    }
    
    func send(event: String, body: [AnyHashable: Any]) {
        eventSink?(["event": event, "body": body])
    }
    
}
